kvserver: TestReplicaProbeRequest  flaked on master

See flake [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=8705417&problemId=4141).

Jira issue: CRDB-24537

```
Failed
=== RUN   TestReplicaProbeRequest
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaProbeRequest723466813
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_probe_test.go:137: ignoring: result is ambiguous: unable to determine whether command was applied via snapshot
    replica_probe_test.go:150: condition failed to evaluate within 45s: waiting for stores to apply command: 3/2
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicaProbeRequest723466813
--- FAIL: TestReplicaProbeRequest (47.62s)
```

--- a/pkg/kv/kvserver/replica_probe_test.go
+++ b/pkg/kv/kvserver/replica_probe_test.go
@@ -125,20 +125,22 @@ func TestReplicaProbeRequest(t *testing.T) {
 			kvpb.RoutingPolicy_LEASEHOLDER,
 			kvpb.RoutingPolicy_NEAREST,
 		} {
-			var b kv.Batch
-			b.AddRawRequest(probeReq)
-			b.Header.RoutingPolicy = policy
-			err := db.Run(ctx, &b)
-			if errors.HasType(err, (*kvpb.AmbiguousResultError)(nil)) {
-				// Rare but it can happen that we're proposing on a replica
-				// that is just about to get a snapshot. In that case we'll
-				// get:
-				//
-				// result is ambiguous: unable to determine whether command was applied via snapshot
-				t.Logf("ignoring: %s", err)
-				err = nil
-			}
-			require.NoError(t, err)
+			testutils.SucceedsSoon(t, func() error {
+				var b kv.Batch
+				b.AddRawRequest(probeReq)
+				b.Header.RoutingPolicy = policy
+				err := db.Run(ctx, &b)
+				if errors.HasType(err, (*kvpb.AmbiguousResultError)(nil)) {
+					// Rare but it can happen that we're proposing on a replica
+					// that is just about to get a snapshot. In that case we'll
+					// get:
+					//
+					// result is ambiguous: unable to determine whether command was applied via snapshot
+					return errors.Wrapf(err, "retrying")
+				}
+				require.NoError(t, err)
+				return nil
+			})
 		}
 	}
 	// Check expected number of probes seen on each Replica in the apply loop.

